pkg/registry/service/allocator/etcd/etcd.go

```diff
@@ -31,7 +30,7 @@ import (
 )
 
 var (
-	errorUnableToAllocate = errors.New("unable to allocate")
+	errorUnableToAllocate = k8serr.NewInternalError(fmt.Errorf("unable to allocate"))
```

Comment:
It will still be converted to an InternalError, even if we do not generate one here.
The advantage of generating an InternalError here is that we can then compare that the returned error == errorUnableToAllocate.
Otherwise, we will have to do
error.Error() == NewInternalError(errorUnableToAllocate).Error()